crates/slightfile/src/lib.rs

```diff
@@ -0,0 +1,322 @@
+use anyhow::{bail, Result};
+use std::{collections::HashMap, fmt::Display, path::Path};
+
+use serde::{Deserialize, Serialize};
+
+/// slightfile version.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub enum SpecVersion {
+    /// Version 0.1 format.
+    #[serde(rename = "0.1")]
+    V1,
+    /// Version 0.2 format.
+    #[serde(rename = "0.2")]
+    V2,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct TomlFile {
+    pub specversion: SpecVersion,
+    pub secret_store: Option<SecretStoreResource>,
+    pub secret_settings: Option<Vec<Config>>,
+    pub capability: Option<Vec<Capability>>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Capability {
+    V1(CapabilityV1),
+    V2(CapabilityV2),
+}
+
+impl Capability {
+    pub fn is_v1(&self) -> bool {
+        matches!(self, Capability::V1(_))
+    }
+    pub fn is_v2(&self) -> bool {
+        matches!(self, Capability::V2(_))
+    }
+    pub fn resource(&self) -> Resource {
+        match self {
+            Capability::V1(c) => c.name.clone(),
+            Capability::V2(c) => c.resource.clone(),
+        }
+    }
+    pub fn name(&self) -> String {
+        match self {
+            Capability::V1(c) => c.name.to_string(),
+            Capability::V2(c) => c.name.clone(),
+        }
+    }
+    pub fn configs(&self) -> Option<HashMap<String, String>> {
+        match self {
+            Capability::V1(_) => None,
+            Capability::V2(c) => c.configs.clone(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct CapabilityV1 {
+    pub name: Resource,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct CapabilityV2 {
+    pub resource: Resource,
+    pub name: String,
+    pub configs: Option<HashMap<String, String>>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Config {
+    pub name: String,
+    pub value: String,
+}
+
+impl Config {
+    pub fn new(name: String, value: String) -> Self {
+        Self { name, value }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum SecretStoreResource {
+    #[serde(rename = "configs.azapp")]
+    Azapp,
+    #[serde(rename = "configs.envvars")]
+    Envvars,
+    #[serde(rename = "configs.usersecrets")]
+    Usersecrets,
+    #[serde(rename = "configs.local")]
+    Local,
+}
+
+impl TryFrom<String> for SecretStoreResource {
+    type Error = anyhow::Error;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        match value.as_str() {
+            "configs.azapp" => Ok(SecretStoreResource::Azapp),
+            "configs.envvars" => Ok(SecretStoreResource::Envvars),
+            "configs.usersecrets" => Ok(SecretStoreResource::Usersecrets),
+            "configs.local" => Ok(SecretStoreResource::Local),
```

Comment:
Is there a better way to do this? I think making people that add new capabilities have to come modify this is a bit of an annoying indirection.

Reply:
I don't know there is a better way to do this... I agree that people who add new capabilities will need to come here to modify slightfile but they need to do that anyway in slight CLI and capability crate.